internal/common/physics/engine.go

```diff
@@ -32,8 +25,8 @@ func (eng *Engine) CalcVelocity(acc float64, dt float64, currVelocity float64) f
 	return newVelocity
 }
 
-func (eng *Engine) Update_loc(coordinates Coordinates, velocity float64, Orientation float64) Coordinates {
-	coordinates.X += velocity * float64(math.Cos(float64(math.Pi*Orientation)))
-	coordinates.Y += velocity * float64(math.Sin(float64(math.Pi*Orientation)))
+func (eng *Engine) UpdateLoc(coordinates utils.Coordinates, velocity float64, orientation float64) utils.Coordinates {
```

Comment:
just one minor thing: I would call this something else (like GetNewLoc or something) as it doesn't actually update the location and the actual UpdateLoc function will be part of the EnvironmentObject interface instead